### PR TITLE
fix:Eliminated redundant password hashing

### PR DESCRIPTION
**Pull request by leoriczhao**
Fix: Eliminated redundant password hashing

Previously, the password was being hashed multiple times upon user registration or update, which was unnecessary . Addressed this by ensuring the password is hashed only once before being stored in the database.

### CODE REVIEW

**Comment by libreliu** [comment]
Thanks for your contribution! May I check if the current implementation (before the PR) works or not?
If there had redundant hashing, newly registered users may not enter the system since the login process would do the same..?

**Comment by libreliu** [comment]
Let me rephrase if I'm not clear enough:

The sign up process requires the raw password to be sent to server, and in `UserSignupView` the password got hashed and then stored in the database. When authenticating users, the view `UserLoginView` calls `UserLoginSerializer`, which then authenticate users using `django.contrib.auth.authenticate` internally. I'm not sure why "the password was being hashed multiple times upon user registration or update". Can you explain this to me..?

**Comment by ZelongZhao** [comment]
The raw password got hashed in `UserSignupView` ,
```Python
def post(self, request, *args):
        # 注册
        try:
            data = request.data.copy()
            data['password'] = make_password(data['password'])
            serializer = UserSerializer(data=data)
            serializer.is_valid(True)
            serializer.save()
            return Response('注册成功', status.HTTP_201_CREATED)
        except Exception as e:
            return Response(str(e), status.HTTP_500_INTERNAL_SERVER_ERROR)
```
when the data is new data that is not associated with an existing database record, so .save() calls .create(). 

```Python
    def create(self, validated_data):
        validated_data['password'] = make_password(validated_data.get('password'))
        try:
            validate_password(validated_data['password'])
            validated_data['is_password_strong'] = True
        except ValidationError:
            validated_data['is_password_strong'] = False

        return super(UserSerializer, self).create(validated_data)
```
than the password was being hashed multiple times @libreliu

**Comment by ZelongZhao** [comment]
Calling will either create a new instance, or update an existing instance, depending on if an existing instance was passed when instantiating the serializer class:.save()
```Python
# .save() will create a new instance.
serializer = CommentSerializer(data=data)

# .save() will update the existing `comment` instance.
serializer = CommentSerializer(comment, data=data)
```

This is what I excerpted from django-rest-framework's API Guide

**Comment by libreliu** [comment]
I see. This seems to be a regression and the problem is caused by my recent commit bd3e7e9. Thanks for pointing out!

**Comment by libreliu** [comment]
Updated, and the regression is now fixed. Thanks again for pointing out!

**Comment by ZelongZhao** [comment]
@libreliu I have registered a lot of accounts, can u help me to delete them.The accounts all  registered with my email:leoric@vfeda.com

**Comment by libreliu** [comment]
> @libreliu I have registered a lot of accounts, can u help me to delete them.The accounts all registered with my email:[leoric@vfeda.com](mailto:leoric@vfeda.com)

Practically nothing bad will happen if these accounts are left intact, and currently limited filter mechanisms are implemented in administration panel, so these deletes will have to be done by hand in Django shell. I prefer not to touch those redundant accounts at this moment. Apologize for the inconvenience and thank you for your understanding.

**Comment by ZelongZhao** [comment]
Thank you for the explanation. It makes sense to keep these accounts as they are for now. I appreciate your hard work and patience in handling this matter.
